src/main/kotlin/game/racing/domain/RacingGame.kt

```diff
@@ -1,21 +1,22 @@
 package game.racing.domain
 
+import game.racing.util.getRandomValue
 import game.racing.view.InputView
 import game.racing.view.ResultView
 
-class RacingGame(private val carManger: CarManger) {
+class RacingGame(private val racingManager: RacingManager, private val tryNumber: Int) {
     fun start() {
-        val tryNumber = InputView.inputTryNumber()
-
         ResultView.printResultIntroduction()
         repeat(tryNumber) {
-            carManger.moveCars()
-            ResultView.printResult(carManger.cars)
+            racingManager.moveCars(List(racingManager.cars.size) { getRandomValue() })
```

Comment:
결국 random값을 cars 수만큼 실행하도록하면 될 것 같은데, 지금처럼 `getRandomValue()`를 최상위 함수로 둔다면 복잡한 구현이 될 것 같아요. 🤔

만약 RacingManager의 `moveCars` 메서드를 아래와 같이 변경한다면 
```kotlin
    fun moveCars(moveStrategy :() -> Int) {
        cars.forEach {
            it.moveOrStayBySpeed(moveStrategy())
        }
    }

```

해당 로직을 아래와 같이 변경해도 동일하게 동작시킬 수 있을 것 같아요.
```suggestion
            racingManager.moveCars{ getRandomValue() }
```

Reply:
와 함수를 인자로 넘길 생각을 못했네요... 덕분에 훨씬 코드 가독성이 좋아졌네요 감사합니다.